packages/block-editor/src/hooks/fit-text.js

```diff
@@ -60,7 +66,54 @@ function addAttributes( settings ) {
  */
 function useFitText( { fitText, name, clientId } ) {
 	const hasFitTextSupport = hasBlockSupport( name, FIT_TEXT_SUPPORT_KEY );
-	const blockElement = useBlockElement( clientId );
+	const refBlockElement = useBlockElement( clientId );
+
+	// Fallback: Try to query the DOM directly when refBlockElement is null
+	// This is needed because in preview mode useBlockElement always returns null
+	// that probably is an existing bug but the fix is not obvious so for now lets
+	// try this workaround on fit text only.
+	const [ fallbackElement, setFallbackElement ] = useState( null );
+	useLayoutEffect( () => {
```

Comment:
This block painting, but the operation is cheap (finding a couple of DOM nodes), so it should be fine.